### PR TITLE
added check in importMulti modal to add multi-file dataset as single …

### PR DESCRIPTION
**Pull request by jackgrossman18**
ref #1316

### CODE REVIEW

**Comment by jackgrossman18** [comment]
@brianhatchl @maxgrossman this is ready for review

**Comment by brianhatchl** [comment]
bruh :eyes: :fearful:

**Comment by maxgrossman** [comment]
You can blame force pushes on me. also emoji less please.

I noticed that 2 of the previous hoot2x merges were brining in stale versions of the hoot2x branch (perhaps because jack's local hoot2x was out of sync, I'm not sure).

So I just suggested to roll back to the previous commit before those bad merges, cherry-pick the good commit after them and push to branch.

**Comment by maxgrossman** [comment]
Though, I guess I should have made a point to do on a separate branch - that is my bad.

**Comment by brianhatchl** [comment]
There was no :melvin: but I guess :neckbeard: is close enough.
Just wanted to make sure we almost never resort to using force push.
This workflow is important.
Whenever I merge I make sure to do a `git fetch` and then `git merge origin/<branch_name>` so it doesn't matter if my local branch has pulled.

**Comment by jackgrossman18** [comment]
ultimately it's my fault, I'm working through getting the progress bar behavior you asked for
:hammer_and_wrench:

**Comment by maxgrossman** [comment]
:muscle:

**Comment by jackgrossman18** [comment]
@brianhatchl i updated the progress bar process so the success css behavior is the same for both uploading multiple layers and combining two layers as a single layer

File: modules/Hoot/ui/modals/ImportMultiDatasets.js [hunk 1]
```diff
@@ -348,32 +365,30 @@ export default class ImportMultiDatasets {
...
-                    formData: this.getFormData( importFiles ),
+                    INPUT_NAME: asSingleName,
+                    formData: this.getFormData(this.fileIngest.node().files),
                     folderId
                 };
 
                 Hoot.api.uploadDataset( params )
                     .then( resp => Hoot.api.statusInterval( resp.data[ 0 ].jobid ) )
+                    .then( () => {
+                        this.progressBar.property('value', fileNames.length );
```
Comment (brianhatchl) [inline modules/Hoot/ui/modals/ImportMultiDatasets.js]: just increment current value by 1 when a promise resolves.

File: modules/Hoot/ui/modals/ImportMultiDatasets.js [hunk 2]
```diff
@@ -387,41 +402,73 @@ export default class ImportMultiDatasets {
...
+
+                    Hoot.api.uploadDataset( params )
+                        .then( resp => Hoot.api.statusInterval( resp.data[ 0 ].jobid ) )
+                        .then( () => {
+                            this.progressBar.property( 'value', index + 1 );
+                            this.fileListInput
+                                .select( `option[value="${name}"]` )
+                                .classed( 'import-success', true );
+                        } )
+                        .then( () => resolve( name ) )
```
Comment (brianhatchl) [inline modules/Hoot/ui/modals/ImportMultiDatasets.js]: Not sure resolve is needed any longer as you've moved the callback code (updating progress bar and filename highlighting) directly into the promise.

However, I think something can be done to make this section of code more DRY.  The async method in both the asSingle and normal multiple upload are almost identical except for the filtering of files into each set to upload and the import-success selector.  I think we could just set a variable to skip the filtering of importFiles and the `[value=]' css selector

**Pull request merged**